File: Dreams/DataDirectory.vb
Imports System.IO
Imports System.Xml

Public Class DataDirectory

    Public Shared Function GetFolderPath() As String
        Dim strPath As String = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) + "\lucidcode\"
        If (File.Exists(strPath & "LightenedDreamDirectory.ld3")) Then
            Dim xmlDocDataDirectory As New XmlDocument
            xmlDocDataDirectory.Load(strPath & "LightenedDreamDirectory.ld3")
            If Directory.Exists(xmlDocDataDirectory.DocumentElement.SelectSingleNode("Directory").InnerText) Then
                Return xmlDocDataDirectory.DocumentElement.SelectSingleNode("Directory").InnerText
            End If
        End If

        Return Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) + "\lucidcode\"
    End Function

End Class

File: Categories/DataDirectory.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.IO;
using System.Xml;
namespace lucidcode.LightenedDream.Categories
{

	public class DataDirectory
	{

		public static string GetFolderPath()
		{
			string strPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) + "\\lucidcode\\";
			if ((File.Exists(strPath + "LightenedDreamDirectory.ld3"))) {
				XmlDocument xmlDocDataDirectory = new XmlDocument();
				xmlDocDataDirectory.Load(strPath + "LightenedDreamDirectory.ld3");
				if (Directory.Exists(xmlDocDataDirectory.DocumentElement.SelectSingleNode("Directory").InnerText)) {
					return xmlDocDataDirectory.DocumentElement.SelectSingleNode("Directory").InnerText;
				}
			}

			return Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) + "\\lucidcode\\";
		}

	}
}
